suite-native/alerts/src/components/AlertSheet.tsx

```diff
@@ -50,42 +36,44 @@ const sheetOverlayStyle = prepareNativeStyle(_ => ({
     flex: 1,
 }));
 
-export const AlertSheet = ({
-    isVisible,
-    title,
-    description,
-    primaryButtonTitle,
-    secondaryButtonTitle,
-    onPressPrimaryButton,
-    onPressSecondaryButton,
-}: AlertSheetProps) => {
+export const AlertSheet = ({ alert }: AlertSheetProps) => {
+    const { setAlert } = useAlert();
     const { applyStyle } = useNativeStyles();
     const { runShakeAnimation, shakeAnimatedStyle } = useShakeAnimation();
+
+    const handleHideAlert = () => setAlert(null);
     const {
         animatedSheetWithOverlayStyle,
         animatedSheetWrapperStyle,
         closeSheetAnimated,
         openSheetAnimated,
-    } = useBottomSheetAnimation({ isVisible });
+    } = useBottomSheetAnimation({ onClose: handleHideAlert, isVisible: true });
 
     useEffect(() => {
-        if (isVisible) {
-            openSheetAnimated();
-        }
-    }, [isVisible, openSheetAnimated]);
+        openSheetAnimated();
+    }, [openSheetAnimated]);
+
+    const {
+        title,
+        description,
+        onPressPrimaryButton,
+        primaryButtonTitle,
+        onPressSecondaryButton,
+        secondaryButtonTitle,
+    } = alert;
 
-    const handlePressPrimaryButton = () => {
+    const handlePressPrimaryButton = async () => {
+        await closeSheetAnimated();
         onPressPrimaryButton();
-        closeSheetAnimated();
     };
 
-    const handlePressSecondaryButton = () => {
+    const handlePressSecondaryButton = async () => {
+        await closeSheetAnimated();
         onPressSecondaryButton?.();
-        closeSheetAnimated();
     };
 
     return (
-        <Modal transparent visible={isVisible}>
+        <Modal transparent visible>
```

Comment:
Animation handles visibility now, or is this a mistake?

Reply:
Modal needs to be visible all the time when an alert is stored in the state. If it would stay as `visible={isVisible}` then the modal would disappear before the animation would finish. On the contrary, if the `visible` would be completely removed, then to content is never displayed.

Reply:
This is very confusing we need to solve it somehow better. Maybe `useBottomSheetAnimation` could return some bool that will be set after animation finish?

Reply:
fixed in 5bdfc744e